rhino/src/main/java/org/mozilla/classfile/ClassFileWriter.java

```diff
@@ -1456,7 +1455,8 @@ private SuperBlock getSuperBlockFromOffset(int offset) {
             if (startIdx < itsSuperBlockStartsTop) {
                 SuperBlock sb = superBlocks[startIdx];
                 // check, if it is really the matching one
-                if (offset < sb.getStart() || offset >= sb.getEnd()) Kit.codeBug();
+                if (offset < sb.getStart() || offset >= sb.getEnd())
+                    throw new IndexOutOfBoundsException();
```

Comment:
please add a message with at least the current range and the wrong index that was used

Reply:
Thanks for the hint. I've made also a separate PR where I've fixed some other places (see #1892)